drivers/clock_management/nxp_syscon/nxp_syscon_gate.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/clock_management/clock_driver.h>
+
+#define DT_DRV_COMPAT nxp_syscon_clock_gate
+
+struct syscon_clock_gate_config {
+	const struct clk *parent;
+	volatile uint32_t *reg;
+	uint8_t enable_offset;
+};
+
+static int syscon_clock_gate_get_rate(const struct clk *clk_hw)
+{
+	const struct syscon_clock_gate_config *config = clk_hw->hw_data;
+
+	return ((*config->reg) & BIT(config->enable_offset)) ?
+		clock_get_rate(config->parent) : 0;
+}
+
+static int syscon_clock_gate_configure(const struct clk *clk_hw, const void *data)
+{
+	const struct syscon_clock_gate_config *config = clk_hw->hw_data;
+	int parent_rate = clock_get_rate(config->parent);
+	int cur_rate = ((*config->reg)  & BIT(config->enable_offset)) ?
+		parent_rate : 0;
+	int ret;
+	bool ungate = (bool)data;
+
+	if (ungate) {
+		ret = clock_children_check_rate(clk_hw, parent_rate);
+		if (ret < 0) {
+			return ret;
+		}
+		ret = clock_children_notify_pre_change(clk_hw, cur_rate,
+						       parent_rate);
+		if (ret < 0) {
+			return ret;
+		}
+		(*config->reg) |= BIT(config->enable_offset);
```

Comment:
shouldn't this be made in to atomic otherwise you could have a potential race condition here with the RMW